parking_permits/views.py

```diff
@@ -535,12 +535,25 @@ def post(self, request, format=None):
         talpa_subscription_id = request.data.get("subscriptionId")
         event_type = request.data.get("eventType")
 
-        # Always bypass Order cancelled event
-        if event_type == "ORDER_CANCELLED":
-            return ok_response(f"Order {talpa_order_id} cancel bypassed")
-
         if not talpa_order_id:
             return bad_request_response("Talpa order id is missing from request data")
+
+        if event_type == "ORDER_CANCELLED":
+            logger.info(f"Cancelling order: {talpa_order_id}")
+            try:
+                order = Order.objects.get(talpa_order_id=talpa_order_id)
+                order.status = OrderStatus.CANCELLED
+                order.save()
+                for permit in order.permits.all():
```

Comment:
Could this not be done with a single `update()` call?

```python
order.permits.update(status=ParkingPermitStatus.CANCELLED)
```

Reply:
Good point, yes.